master/internal/api_workspace.go

```diff
@@ -1076,23 +1076,21 @@ func (a *apiServer) SetResourceQuotas(ctx context.Context,
 		return nil, err
 	}
 
-	tx, err := db.Bun().BeginTx(ctx, nil)
+	w, err := a.GetWorkspaceByID(ctx, req.Id, *curUser, false)
 	if err != nil {
 		return nil, err
 	}
-	defer func() {
-		if err = tx.Rollback(); err != nil && err != sql.ErrTxDone {
```

Comment:
yeah, it is super weird. It works if I just do defer tx.Rollback() but errors out if I have a defer func(). We usually defer to make sure we cover all our bases with an early return but since here only setResourceQuotas needs a transaction, I think it is okay to just explicitly call the Rollback() function once

Reply:
Resolved! Fix merged [here](https://github.com/determined-ai/determined/pull/9713)